scipy/cluster/hierarchy.py

```diff
@@ -619,55 +618,53 @@ def linkage(y, method='single', metric='euclidean'):
     Z : ndarray
         The hierarchical clustering encoded as a linkage matrix.
 
+    Notes
+    -----
+    1. For method 'single' an optimized algorithm called SLINK is implemented,
+       which has :math:`O(n^2)` time complexity and :math:`O(n)` memory
```

Comment:
Right, I knew this, I was implying "additional memory". Is it better to name it "additional memory" or maybe say that all algorithms have O(n^2) for memory in our implementation?